src/number_theory/padics/padic_norm.lean

```diff
@@ -215,13 +227,28 @@ protected lemma defn {q : ℚ} {n d : ℤ} (hqz : q ≠ 0) (qdf : q = n /. d) :
 have hn : n ≠ 0, from rat.mk_num_ne_zero_of_ne_zero hqz qdf,
 have hd : d ≠ 0, from rat.mk_denom_ne_zero_of_ne_zero hqz qdf,
 let ⟨c, hc1, hc2⟩ := rat.num_denom_mk hn hd qdf in
-by rw [padic_val_rat, dif_pos];
+begin
   simp [hc1, hc2, multiplicity.mul' (nat.prime_iff_prime_int.1 p_prime.1),
-    (ne.symm (ne_of_lt p_prime.1.one_lt)), hqz]
+    (ne.symm (ne_of_lt p_prime.1.one_lt)), hqz],
```

Comment:
What is the reason that this proof exploded?

Reply:
The previous definition of `padic_val_rat` had big proofs of domain membership in the definition. I've readded what was essentially the old definition as a lemma to do this more efficiently.